### PR TITLE
safely delete stack

### PR DESCRIPTION
**Pull request by szuecs**
fix for #68 : implement deleteScheduled tag to postpone deletion of ALBs 1 hour after trigger

to be discussed with @njuettner

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

File: aws/cf.go [hunk 1]
```diff
@@ -36,6 +42,47 @@ func (s *Stack) TargetGroupARN() string {
...
+	}
+	_, ok := s.tags[deleteScheduled]
+	return ok
+}
+
+// ShouldDelete returns true if stack is marked to delete and the
+// deleteScheduled tag is after time.Now(). In all other cases it
+// returns false.
+func (s *Stack) ShouldDelete() bool {
+	if s == nil {
```
Comment (szuecs) [inline aws/cf.go]: just for nicer Go, you can then have a nil ptr to Stack and you can call ShouldDelete() without checking for nil before

File: aws/cf.go [hunk 2]
```diff
@@ -36,6 +42,47 @@ func (s *Stack) TargetGroupARN() string {
...
+	t0 := s.deleteTime()
+	if t0 == nil {
+		return false
+	}
+	now := time.Now()
+	return now.After(*t0)
+}
+
+func (s *Stack) deleteTime() *time.Time {
+	if s == nil {
```
Comment (szuecs) [inline aws/cf.go]: same as above

File: aws/adapter.go [hunk 3]
```diff
@@ -250,11 +253,26 @@ func (a *Adapter) CreateStack(certificateARN string) (string, error) {
...
 // GetStack returns the CloudFormation stack details with the name or ID from the argument
 func (a *Adapter) GetStack(stackID string) (*Stack, error) {
 	return getStack(a.cloudformation, stackID)
 }
 
+// MarkToDeleteStack adds a "deleteScheduled" Tag to the CloudFormation stack with the given name
+func (a *Adapter) MarkToDeleteStack(stack *Stack) (time.Time, error) {
+	t0 := time.Now().Add(a.stackTTL)
+
+	if err := markToDeleteStack(a.cloudformation, a.stackName(stack.CertificateARN()), t0.Format(time.RFC3339)); err != nil {
```
Comment (szuecs) [inline aws/adapter.go]: done

**Comment by Raffo** [comment]
👍

**Comment by Raffo** [comment]
👍

**Comment by szuecs** [comment]
:+1:

**Pull request merged**